Also, the CI Build logs raise the following errors:

```
./homeassistant/components/switch/tplink.py:3:80: E501 line too long (83 > 79 characters)
./homeassistant/components/switch/tplink.py:3:84: W291 trailing whitespace
./homeassistant/components/switch/tplink.py:22:1: E302 expected 2 blank lines, found 1
./homeassistant/components/switch/tplink.py:31:1: E302 expected 2 blank lines, found 1

************* Module homeassistant.components.switch.tplink
C:  3, 0: Trailing whitespace (trailing-whitespace)
C: 22, 0: Missing function docstring (missing-docstring)
E: 23, 4: Unable to import 'pyHS100' (import-error)
W: 24, 4: String statement has no effect (pointless-string-statement)
```

It also raises some other errors which can be ignored. Looks like our linting application got an update.

--- a/.coveragerc
+++ b/.coveragerc
@@ -222,6 +222,7 @@ omit =
     homeassistant/components/switch/pulseaudio_loopback.py
     homeassistant/components/switch/rest.py
     homeassistant/components/switch/rpi_rf.py
+    homeassistant/components/switch/tplink.py
     homeassistant/components/switch/transmission.py
     homeassistant/components/switch/wake_on_lan.py
     homeassistant/components/thermostat/eq3btsmart.py

--- a/homeassistant/components/switch/tplink.py
+++ b/homeassistant/components/switch/tplink.py
@@ -1,7 +1,7 @@
 """Support for TPLink HS100/HS110 smart switch.
 
-It is able to monitor current switch status, as well as turn on and off the switch. 
-
+It is able to monitor current switch status, as well as turn on and off
+the switch. Will work with both HS100 and HS110 switch models.
 """
 
 import logging
@@ -12,21 +12,21 @@ from homeassistant.const import (
 
 # constants
 DEVICE_DEFAULT_NAME = 'HS100'
-REQUIREMENTS = ['http://github.com/gadgetreactor/pyHS100/archive/'
-                '27dd078e57628e76a1de89cc9d81b703f8e846d1'
-                '#pyHS100==0.1.0']
+REQUIREMENTS = ['https://github.com/gadgetreactor/pyHS100/archive/'
+                'master.zip#pyHS100==0.1.2']
 
-# setup logger
-_LOGGER = logging.getLogger(__name__)
+# pylint: disable=unused-argument
+
 
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
-    import pyHS100
+    from pyHS100.pyHS100 import SmartPlug
     """Setup the TPLink platform in configuration.yaml."""
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME, DEVICE_DEFAULT_NAME)
 
-    add_devices_callback([SmartPlugSwitch(pyHS100.SmartPlug(host),
+    add_devices_callback([SmartPlugSwitch(SmartPlug(host),
                                           name)])
+
 
 class SmartPlugSwitch(SwitchDevice):
     """Representation of a TPLink Smart Plug switch."""

--- a/homeassistant/components/switch/tplink.py
+++ b/homeassistant/components/switch/tplink.py
@@ -4,8 +4,6 @@ It is able to monitor current switch status, as well as turn on and off
 the switch. Will work with both HS100 and HS110 switch models.
 """
 
-import logging
-
 from homeassistant.components.switch import SwitchDevice
 from homeassistant.const import (
     CONF_HOST, CONF_NAME)
@@ -19,8 +17,8 @@ REQUIREMENTS = ['https://github.com/gadgetreactor/pyHS100/archive/'
 
 
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
-    from pyHS100.pyHS100 import SmartPlug
     """Setup the TPLink platform in configuration.yaml."""
+    from pyHS100.pyHS100 import SmartPlug
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME, DEVICE_DEFAULT_NAME)
 

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -114,7 +114,7 @@ https://github.com/TheRealLink/pythinkingcleaner/archive/v0.0.2.zip#pythinkingcl
 https://github.com/Xorso/pyalarmdotcom/archive/0.1.1.zip#pyalarmdotcom==0.1.1
 
 # homeassistant.components.media_player.braviatv
-https://github.com/aparraga/braviarc/archive/0.3.2.zip#braviarc==0.3.2
+https://github.com/aparraga/braviarc/archive/0.3.3.zip#braviarc==0.3.3
 
 # homeassistant.components.media_player.roku
 https://github.com/bah2830/python-roku/archive/3.1.2.zip#roku==3.1.2
@@ -292,7 +292,7 @@ pynetio==0.1.6
 pynx584==0.2
 
 # homeassistant.components.sensor.openweathermap
-pyowm==2.3.1
+pyowm==2.3.2
 
 # homeassistant.components.switch.acer_projector
 pyserial<=3.0
@@ -320,7 +320,7 @@ python-nmap==0.6.0
 python-pushover==0.2
 
 # homeassistant.components.notify.telegram
-python-telegram-bot==4.3.1
+python-telegram-bot==4.3.2
 
 # homeassistant.components.sensor.twitch
 python-twitch==1.2.0
@@ -366,7 +366,7 @@ scsgate==0.1.0
 sendgrid>=1.6.0,<1.7.0
 
 # homeassistant.components.notify.slack
-slacker==0.9.18
+slacker==0.9.21
 
 # homeassistant.components.notify.xmpp
 sleekxmpp==1.3.1
@@ -433,5 +433,11 @@ xbee-helper==0.0.7
 # homeassistant.components.sensor.yr
 xmltodict
 
+# homeassistant.components.sensor.yweather
+yahooweather==0.4
+
 # homeassistant.components.zeroconf
 zeroconf==0.17.5
+
+# homeassistant.components.switch.tplink
+https://github.com/gadgetreactor/pyHS100/archive/master.zip#pyHS100==0.1.2

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -114,7 +114,7 @@ https://github.com/TheRealLink/pythinkingcleaner/archive/v0.0.2.zip#pythinkingcl
 https://github.com/Xorso/pyalarmdotcom/archive/0.1.1.zip#pyalarmdotcom==0.1.1
 
 # homeassistant.components.media_player.braviatv
-https://github.com/aparraga/braviarc/archive/0.3.3.zip#braviarc==0.3.3
+https://github.com/aparraga/braviarc/archive/0.3.2.zip#braviarc==0.3.2
 
 # homeassistant.components.media_player.roku
 https://github.com/bah2830/python-roku/archive/3.1.2.zip#roku==3.1.2
@@ -127,6 +127,9 @@ https://github.com/danieljkemp/onkyo-eiscp/archive/python3.zip#onkyo-eiscp==0.9.
 
 # homeassistant.components.device_tracker.fritz
 # https://github.com/deisi/fritzconnection/archive/b5c14515e1c8e2652b06b6316a7f3913df942841.zip#fritzconnection==0.4.6
+
+# homeassistant.components.switch.tplink
+https://github.com/gadgetreactor/pyHS100/archive/master.zip#pyHS100==0.1.2
 
 # homeassistant.components.netatmo
 https://github.com/jabesq/netatmo-api-python/archive/v0.5.0.zip#lnetatmo==0.5.0
@@ -292,7 +295,7 @@ pynetio==0.1.6
 pynx584==0.2
 
 # homeassistant.components.sensor.openweathermap
-pyowm==2.3.2
+pyowm==2.3.1
 
 # homeassistant.components.switch.acer_projector
 pyserial<=3.0
@@ -320,7 +323,7 @@ python-nmap==0.6.0
 python-pushover==0.2
 
 # homeassistant.components.notify.telegram
-python-telegram-bot==4.3.2
+python-telegram-bot==4.3.1
 
 # homeassistant.components.sensor.twitch
 python-twitch==1.2.0
@@ -366,7 +369,7 @@ scsgate==0.1.0
 sendgrid>=1.6.0,<1.7.0
 
 # homeassistant.components.notify.slack
-slacker==0.9.21
+slacker==0.9.18
 
 # homeassistant.components.notify.xmpp
 sleekxmpp==1.3.1
@@ -433,11 +436,5 @@ xbee-helper==0.0.7
 # homeassistant.components.sensor.yr
 xmltodict
 
-# homeassistant.components.sensor.yweather
-yahooweather==0.4
-
 # homeassistant.components.zeroconf
 zeroconf==0.17.5
-
-# homeassistant.components.switch.tplink
-https://github.com/gadgetreactor/pyHS100/archive/master.zip#pyHS100==0.1.2

